src/site/content/en/blog/read-files/index.md

```diff
@@ -0,0 +1,253 @@
+
+```html
+<!-- The `multiple` attribute lets users select multiple files. -->
+<input type="file" id="input" multiple/>
+<script>
+  if (window.FileList && window.File) {
+    document.getElementById('input').addEventListener('change', event => {
+      const files = event.target.files; // This is a `FileList` object.
+      // `forEach()` won't work unless you convert `files` to an `Array`.
+      for (let i = 0; i < files.length; i++) {
+        const file = files[i]; // This is a `File` object.
+        const name = file.name ? file.name : 'NOT SUPPORTED';
+        const type = file.type ? file.type : 'NOT SUPPORTED';
+        console.log({name, type});
+      }
+    }); 
+  }
+</script>
+```
+
+This example lets a user select multiple files using their operating system's built-in
+file selection UI and then logs out the name and file type of each selected file to the Console.
+
+[`<input type="file">`][input] is the recommended way to enable users to select files because
+browser vendors do a lot of work to make sure that their built-in elements are accessible.
+Some browsers also treat `<input type="file">` elements as automatic drag-and-drop targets.
+After the user clicks the `<input type="file">` element the operating system's built-in
+file selection UI appears. Add the `multiple` attribute to your `<input type="file">` element
+to instruct the operating system to allow the user to select multiple files. When the
+user finishes selecting a file or files the browser fires the `<input type="file">` element's
+`change` event. You can access the list of files from `event.target.files`, which is a [`FileList`][filelist]
+object, not an `Array`. If you want to iterate over `files` with `forEach()` you'll need to convert
+it to an `Array`. Each item in the `FileList` is a [`File`][file] object.
+
+<div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe src="https://glitch.com/embed/#!/embed/input-type-file?previewSize=100"
+          alt="How to select local files and read file metadata with JavaScript."
+          style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+Browser compatibility data for the APIs used in this section:
+[`<input type="file">`](https://developer.mozilla.org/docs/Web/HTML/Element/input/file#Browser_compatibility),
+[`File`](https://developer.mozilla.org/docs/Web/API/File#Browser_compatibility),
+[`FileList`](https://developer.mozilla.org/docs/Web/API/FileList#Browser_compatibility)
+
+### Select files with a custom drag-and-drop UI (not recommended) {: #drag-and-drop }
+
+{% Aside 'caution' %}
+  [Selecting files with an `<input type="file">` element](#input) is the recommended
+  approach because browser vendors do a lot of work to ensure that their elements
+  are accessible. However, MDN doesn't currently state how many browsers treat their
+  `<input type="file">` elements as built-in drag-and-drop targets.
+{% endAside %}
```

Comment:
@robdodson it seems like the usual rationale for using built-in elements also applies here? Don't use a custom drag-and-drop because it will take a lot of work to handle all the a11y edge cases.

Reply:
that seems ok to me. I don't know much about accessible drag and drop except that everyone says it's hard/impossible. So having an actual element that someone can just tab to sounds good imo.